### PR TITLE
Move to GitHub actions

### PR DESCRIPTION
**Pull request by MancunianSam**
There are a few changes here:
Create a terraform check workflow which calls `tdr-github-actions` check workflow. This https://github.com/nationalarchives/tdr-github-actions/pull/8 will have to be merged before the checks will work.

Create a terraform apply workflow. This will run on merge to master. It will:
* Run terraform plan and export the output to Cloudwatch. 
* Send  a slack message with the link to the output logs.
* Trigger the job to apply the changes. This will require approval on all environments. 

Once you've viewed the plan in Cloudwatch, you can approve the job in GitHub actions (a link will be posted to Slack) 

I've also added `tdr-terraform-modules` and `tdr-configurations` as submodules rather than checking them out explicitly, it simplifies the workflow slightly. If it's annoying, we can always undo it.

### CODE REVIEW

File: .github/workflows/apply.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        id: plan
+        env:
+          GITHUB_OWNER: nationalarchives
+          GITHUB_TOKEN: ${{ secrets.WORKFLOW_PAT }}
+          TF_VAR_tdr_account_number: ${{ secrets[needs.setup.outputs.account-number-secret] }}
+        run: |
+            terraform init
+            terraform workspace select ${{ github.event.inputs.environment }}
+            pip install boto3
+            terraform plan -no-color -out=out > /dev/null
```
Comment (TomJKing) [inline .github/workflows/apply.yml:51]: Will the "color" plan work on GitHub. It was only set to `no-color` as the Jenkins UI couldn't cope with it

File: README.md [hunk 2]
```diff
@@ -18,23 +18,21 @@ The different modules are used by Terraform workspaces which represent three AWS
...
 
-To start a deployment, run the [TDR Terraform Environments Deploy job in Jenkins][jenkins-job] by clicking 'Build with Parameters' and selecting the environment you want to deploy to. All changes must be deployed first to integration, then staging, then production.
+To start a deployment, run the [TDR Terraform Environments Deploy job in GitHub actions][github-actions-job] by clicking 'Run Workflow' and selecting the environment you want to deploy to. All changes must be deployed first to integration, then staging, then production. 
 
-The deployment will pause when Terraform has determined which changes need to be applied. Review the Terraform plan output by opening the job's console output.
+The deployment will pause when Terraform has determined which changes need to be applied. Review the Terraform plan output by clicking the link provided in the Slack notification. This will be a link to Cloudwatch in the management account so you will need to be logged in to the management AWS account to use this.
```
Comment (TomJKing) [inline README.md:23]: Just a thought, do the junior devs (Thanh and Nigel) have access to the management account logs? If not will need to do something about that.

**Review by TomJKing — COMMENTED** [review]
Looks good. Just a couple of minor comments. Thanks

File: .github/workflows/apply.yml (hunk 1 above)
Comment (MancunianSam) [reply .github/workflows/apply.yml:51]: These logs are being uploaded to Cloudwatch and I don't think Cloudwatch supports colours either so we do need this.

File: README.md (hunk 2 above)
Comment (MancunianSam) [reply README.md:23]: Good point, I'll have a look at the role they're attached to. I can put in a PR for digital services to have a look at if not.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**